python/datafusion/tests/test_aggregation.py

```diff
@@ -39,78 +39,53 @@ def df():
-            f.covar_pop(col_a, col_c),
-            f.covar_samp(col_b, col_c),
-            # f.grouping(col_a),  # No physical plan implemented yet
-            f.max(col_a),
-            f.mean(col_b),
-            f.median(col_b),
-            f.min(col_a),
-            f.sum(col_b),
-            f.stddev(col_a),
-            f.stddev_pop(col_b),
-            f.stddev_samp(col_c),
-            f.var(col_a),
-            f.var_pop(col_b),
-            f.var_samp(col_c),
-        ],
-    )
+    agg_df = df.aggregate([], [agg_expr])
     result = agg_df.collect()[0]
     values_a, values_b, values_c, values_d = df.collect()[0]
+    expected = calc_expected(values_a, values_b, values_c, values_d)
+    np.testing.assert_array_almost_equal(result.column(0), expected)
 
-    assert result.column(0) == pa.array([2], type=pa.uint64())
-    assert result.column(1) == pa.array([4])
-    assert result.column(2) == pa.array([4])
-    # Ref: https://github.com/apache/datafusion-python/issues/777
-    # assert result.column(3) == pa.array([6])
-    assert result.column(4) == pa.array([[4, 4, 6]])
-    np.testing.assert_array_almost_equal(result.column(5), np.average(values_a))
-    np.testing.assert_array_almost_equal(
-        result.column(6), np.corrcoef(values_a, values_b)[0][1]
-    )
-    assert result.column(7) == pa.array([len(values_a)])
-    # Sample (co)variance -> ddof=1
-    # Population (co)variance -> ddof=0
-    np.testing.assert_array_almost_equal(
-        result.column(8), np.cov(values_a, values_b, ddof=1)[0][1]
-    )
-    np.testing.assert_array_almost_equal(
-        result.column(9), np.cov(values_a, values_c, ddof=0)[0][1]
-    )
-    np.testing.assert_array_almost_equal(
-        result.column(10), np.cov(values_b, values_c, ddof=1)[0][1]
-    )
-    np.testing.assert_array_almost_equal(result.column(11), np.max(values_a))
-    np.testing.assert_array_almost_equal(result.column(12), np.mean(values_b))
-    np.testing.assert_array_almost_equal(result.column(13), np.median(values_b))
-    np.testing.assert_array_almost_equal(result.column(14), np.min(values_a))
-    np.testing.assert_array_almost_equal(
-        result.column(15), np.sum(values_b.to_pylist())
-    )
-    np.testing.assert_array_almost_equal(result.column(16), np.std(values_a, ddof=1))
-    np.testing.assert_array_almost_equal(result.column(17), np.std(values_b, ddof=0))
-    np.testing.assert_array_almost_equal(result.column(18), np.std(values_c, ddof=1))
-    np.testing.assert_array_almost_equal(result.column(19), np.var(values_a, ddof=1))
-    np.testing.assert_array_almost_equal(result.column(20), np.var(values_b, ddof=0))
-    np.testing.assert_array_almost_equal(result.column(21), np.var(values_c, ddof=1))
+
+@pytest.mark.parametrize("agg_expr, expected", [
+    (f.approx_distinct(column("b")), pa.array([2], type=pa.uint64())),
+    (f.approx_median(column("b")), pa.array([4])),
+    (f.approx_percentile_cont(column("b"), lit(0.5)), pa.array([4])),
```

Comment:
Should we add in a unit test with the num_centroids feature to demonstrate it's working as expected?